fog/ledger/enclave/impl/src/lib.rs

```diff
@@ -252,41 +265,30 @@ where
             .into_iter()
             .map(|(responder_id, query_response)| {
                 let plaintext_bytes = self.ake.backend_decrypt(&responder_id, &query_response)?;
-                let query_response: CheckKeyImagesResponse =
-                    mc_util_serial::decode(&plaintext_bytes)?;
+                let query_response: ShardKeyImageResponse =
+                    mc_util_serial::deserialize(&plaintext_bytes)?;
 
                 Ok(query_response)
             })
-            .collect::<Result<Vec<CheckKeyImagesResponse>>>()?;
+            .collect::<Result<Vec<_>>>()?;
 
-        let num_blocks = shard_query_responses
-            .iter()
-            .map(|query_response| query_response.num_blocks)
-            .min()
-            .expect("this is only None when the iterator is empty but we early-exit in that case");
-        let global_txo_count = shard_query_responses
-            .iter()
-            .map(|query_response| query_response.global_txo_count)
-            .min()
-            .expect("this is only None when the iterator is empty but we early-exit in that case");
-        let latest_block_version = shard_query_responses
-            .iter()
-            .map(|query_response| query_response.latest_block_version)
-            .max()
-            .expect("this is only None when the iterator is empty but we early-exit in that case");
+        let untrusted_response =
+            merge_untrusted_responses(shard_query_responses.iter().map(|r| &r.untrusted_response));
+        let num_blocks = untrusted_response.processed_block_range.end_block;
```

Comment:
same question regarding `num_blocks` vs `end_block` for the range (and if the range doesn't start with 0, should it be `end_block - start_block`, or no because this is referring to the total number of blocks that have been processed?)

Reply:
two strikes on me possibly not communicating in the code well enough.

I think it's only in two places thus it's not much burden to put a comment on them.
Generally always comment a behavior/assumption feels like something else needs to change
let me know if you think a comment in these 2 locations would suffice or if perhaps a simple helper method of `range_to_num_blocks()` would be better

Reply:
Comment should be sufficient!

Reply:
added code comment